### PR TITLE
Add libsmu recipe

### PR DESCRIPTION
**Pull request by tfcollins**
Signed-off-by: Travis F. Collins <travis.collins@analog.com>

Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ ] Source is from official source
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/libsmu/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    # verify that libraries get installed
+    - test -f $PREFIX/lib/libsmu${SHLIB_EXT}  # [not win]
+    - if not exist %PREFIX%\\Library\\bin\\libsmu.dll exit 1  # [win]
+    - if not exist %PREFIX%\\Library\\lib\\libsmu.lib exit 1  # [win]
+  imports:
+    - pysmu
+
+about:
+  home: https://github.com/analogdevicesinc/libsmu
+  license: BSD
```
Comment (marcelotrevisani) [inline recipes/libsmu/meta.yaml:52]: is that bsd 3? if so, please change it to
```suggestion
  license: BSD-3-Clause
```

File: recipes/libsmu/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  - url: https://github.com/analogdevicesinc/{{ name }}/archive/v{{ version }}.tar.gz
+    sha256: 290d72bf5cec9ceab0030f6a4bc49bfd962aff56997aebebf70a20cb4a6bf77a
+
+build:
+  number: 1
```
Comment (marcelotrevisani) [inline recipes/libsmu/meta.yaml:13]:
```suggestion
  number: 0
```

File: recipes/libsmu/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
+{% set name = "libsmu" %}
+{% set version = "1.0.3" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  - url: https://github.com/analogdevicesinc/{{ name }}/archive/v{{ version }}.tar.gz
+    sha256: 290d72bf5cec9ceab0030f6a4bc49bfd962aff56997aebebf70a20cb4a6bf77a
```
Comment (marcelotrevisani) [inline recipes/libsmu/meta.yaml:10]:
```suggestion
  url: https://github.com/analogdevicesinc/{{ name }}/archive/v{{ version }}.tar.gz
  sha256: 290d72bf5cec9ceab0030f6a4bc49bfd962aff56997aebebf70a20cb4a6bf77a
```

File: recipes/libsmu/bld.bat [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+:: install
+cmake --build . --config Release --target install
+if errorlevel 1 exit 1
+
+:: py install
+cd ..
+cd bindings
+cd python
+python setup.py build_ext -I "%LIBRARY_INC%\libusb-1.0"
+python setup.py install
```
Comment (marcelotrevisani) [inline recipes/libsmu/bld.bat:68]:
```suggestion
%PYTHON% setup.py build_ext -I "%LIBRARY_INC%\libusb-1.0"
%PYTHON% setup.py install
```

File: recipes/libsmu/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+build:
+  number: 1
+  run_exports:
+    - {{ pin_subpackage('libsmu', max_pin='x.x') }}
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - cmake
+    - {{ cdt('libudev-devel') }}  # [linux]
```
Comment (marcelotrevisani) [inline recipes/libsmu/meta.yaml:21]: I believe the ``cdt`` should be in ``host``

**Review by dopplershift — APPROVED** [review]

**Pull request merged**